### PR TITLE
use unified root logger for fx dependency-injection in chaos-controller-manager

### PR DESCRIPTION
**Pull request by STRRL**
### What problem does this PR solve?

It's a part of #2752.

### What's changed and how it works?

- use a certain `logr.Logger` for both controller-runtime and  fx 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: cmd/chaos-controller-manager/main.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ package main
 
 import (
 	"flag"
+	stdlog "log"
 	"net/http"
 	_ "net/http/pprof"
```
[the remark this answers is not in the document]
Comment (STRRL) [reply cmd/chaos-controller-manager/main.go:22]: @sonatype-lift help

[the remark this answers is not in the document]
Comment (STRRL) [reply cmd/chaos-controller-manager/main.go:22]: @sonatype-lift ignore

**Comment by STRRL** [comment]
PTAL

/cc @iguoyr @WangXiangUSTC @YangKeao

File: pkg/log/global_logger.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// Deprecated: Do not use global logger anymore. For more detail, see
+// https://github.com/chaos-mesh/rfcs/blob/main/text/2021-12-09-logging.md#global-logger
+func L() logr.Logger {
+	globalMu.RLock()
+	result := globalLogger
+	globalMu.RUnlock()
+	return result
+}
+
+// ReplaceGlobals would replace the global logger with the given logger.
```
Comment (iguoyr) [inline pkg/log/global_logger.go:47]: Add some comments about this function should only be called when application starting?

Comment (STRRL) [reply pkg/log/global_logger.go:47]: How about "should be called when application starting" ?🤔 I think this restriction ("should only be") is not necessary.

Comment (iguoyr) [reply pkg/log/global_logger.go:47]: Agree with you :)

Comment (STRRL) [reply pkg/log/global_logger.go]: Updated, PTAL!

**Review by iguoyr — APPROVED** [review]
LGTM

**Review by Andrewmatilde — APPROVED** [review]
LGTM. Need example in future.

**Comment by STRRL** [comment]
> cherry pick to release-2.1 failed

I would manually cherry pick it.

**Pull request merged**